# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1297212403 18000
#      Tue Feb 08 19:46:43 2011 -0500
# Node ID 7c60b2bd7f39bd940cf3345af7e39318f463118a
# Parent  a01af8355c463fc92412c3a0345fb7b337d2283e
__go_draw_axes__.m: Properly set fontspec for legends.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2011-02-08  Ben Abbott  <bpabbott@mac.com>
+
+	* plot/__go_draw_axes__.m: Properly set fontspec for legends.
+
 2011-02-08  John W. Eaton  <jwe@octave.org>
 
 	* deprecated/complement.m, deprecated/create_set.m,
 	deprecated/dmult.m, deprecated/iscommand.m,
 	deprecated/israwcommand.m, deprecated/lchol.m,
 	deprecated/loadimage.m, deprecated/mark_as_command.m,
 	deprecated/mark_as_rawcommand.m, deprecated/spatan2.m,
 	deprecated/spchol2inv.m, deprecated/spcholinv.m,
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -1467,17 +1467,17 @@ function __go_draw_axes__ (h, plot_strea
           pos = "";
           warning ("legend: 'Best' not yet implemented for location specifier.\n");
           ## Least conflict with data in plot.
           ## Least unused space outside plot.
         otherwise
           pos = "";
       endswitch
       if (__gnuplot_has_feature__ ("key_has_font_properties"))
-        [fontname, fontsize] = get_fontname_and_size (obj);
+        [fontname, fontsize] = get_fontname_and_size (hlgnd);
         fontspec = create_fontspec (fontname, fontsize, gnuplot_term);
       else
         fontspec = "";
       endif
       fprintf (plot_stream, "set key %s %s;\nset key %s %s %s %s;\n",
                inout, pos, box, reverse, horzvert, fontspec);
     else
       fputs (plot_stream, "unset key;\n");
